File: app/src/main/java/vergecurrency/vergewallet/viewmodel/PaperkeyDistributionViewModel.java
package vergecurrency.vergewallet.viewmodel;

import androidx.lifecycle.ViewModel;
import vergecurrency.vergewallet.service.model.MnemonicManager;
import vergecurrency.vergewallet.service.model.PreferencesManager;
import vergecurrency.vergewallet.wallet.WalletManager;

public class PaperkeyDistributionViewModel extends ViewModel {

    String[] seed;

    public PaperkeyDistributionViewModel() {
        if(!PreferencesManager.isFirstLaunch()) {
            seed = MnemonicManager.getMnemonicFromJSON(PreferencesManager.getMnemonic());
        }

    }

    public String[] getSeed() {

        return seed;
    }

    public void setSeed(String[] seed) {
        this.seed = seed;
    }

    public void generateMnemonics() {
        //Should be created at the launch
        WalletManager.generateMnemonic();
        seed = MnemonicManager.getMnemonicFromJSON(PreferencesManager.getMnemonic());
    }
}

File: app/src/main/kotlin/vergecurrency/vergewallet/viewmodel/PaperkeyDistributionViewModel.kt
package vergecurrency.vergewallet.viewmodel

import androidx.lifecycle.ViewModel
import vergecurrency.vergewallet.service.model.EncryptedPreferencesManager
import vergecurrency.vergewallet.service.model.MnemonicManager
import vergecurrency.vergewallet.service.model.PreferencesManager
import vergecurrency.vergewallet.wallet.WalletManager

class PaperkeyDistributionViewModel : ViewModel() {

    lateinit var seed: Array<String>

    init {
        if (!PreferencesManager.isFirstLaunch) {
            seed = MnemonicManager.getMnemonicFromJSON(EncryptedPreferencesManager.mnemonic!!)!!
        }

    }

    fun generateMnemonics() {
        //Should be created at the launch
        WalletManager.generateMnemonic()
        seed = MnemonicManager.getMnemonicFromJSON(EncryptedPreferencesManager.mnemonic!!)!!
    }
}
